demos/dqx_demo_llm_pk_detection.py

```diff
@@ -0,0 +1,607 @@
+
+# COMMAND ----------
+
+# MAGIC
+# MAGIC %pip install  --no-cache-dir --force-reinstall  databricks_labs_dqx-0.10.0-py3-none-any.whl[llm]
+# MAGIC
+# MAGIC
+
+# COMMAND ----------
+
+# MAGIC
+# MAGIC %restart_python
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Setup and Configuration
+# MAGIC
+# MAGIC ### Model Configuration
+# MAGIC
+# MAGIC The LLM-based primary key detection uses the `databricks-meta-llama-3-1-8b-instruct` foundation model endpoint by default. This is hardcoded specifically for PK detection to ensure optimal performance. Other LLM features in DQX use the global default model (`databricks/databricks-claude-sonnet-4-5`).
+# MAGIC
+# MAGIC ### Table Configuration
+# MAGIC
+# MAGIC You can detect primary keys for any table accessible in your Databricks workspace. Provide the fully qualified table name in the format: `catalog.schema.table`
+
+# COMMAND ----------
+
+# Configuration widgets
+# Note: The model for PK detection is hardcoded to databricks-meta-llama-3-1-8b-instruct
+default_table_name = "samples.tpch.customer"
+
+dbutils.widgets.text("table_name", default_table_name, "Table Name")
+
+table_name = dbutils.widgets.get("table_name")
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+from databricks.labs.dqx.profiler.generator import DQGenerator
+from databricks.labs.dqx.config import LLMModelConfig, InputConfig
+
+# COMMAND ----------
+
+# Initialize Workspace Client
+ws = WorkspaceClient()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Example 1: Simple Primary Key Detection
+# MAGIC
+# MAGIC Let's create a simple table with a clear single-column primary key and detect it using LLM.
+
+# COMMAND ----------
+
+# Create sample data with a clear primary key
+data = [
+    (1, "Alice", "alice@example.com", "Engineering"),
+    (2, "Bob", "bob@example.com", "Sales"),
+    (3, "Charlie", "charlie@example.com", "Marketing"),
+    (4, "David", "david@example.com", "Engineering"),
+    (5, "Eve", "eve@example.com", "HR"),
+]
+
+df = spark.createDataFrame(data, ["user_id", "username", "email", "department"])
+
+# Create temporary table
+temp_table = "dqx_demo_simple_pk"
+df.write.mode("overwrite").saveAsTable(temp_table)
+
+display(df)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Detect Primary Key with LLM
+# MAGIC
+# MAGIC You can detect primary keys using either:
+# MAGIC 1. **DQProfiler** - Convenient method for standalone PK detection
+# MAGIC 2. **DQGenerator** - More control when integrating with rule generation
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC #### Option 1: Using DQProfiler (Recommended for standalone PK detection)
+
+# COMMAND ----------
+
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+# Create profiler
+profiler = DQProfiler(ws, spark)
+
+# Note: PK detection uses databricks-meta-llama-3-1-8b-instruct (hardcoded)
+# LLMModelConfig is not needed for PK detection as the model is preset
+llm_model_config = None
```

Comment:
```suggestion
```
just remove it instead of creating a lot of comments, you can have another example where you define the config. Most users won't care which model do we use, don't need to repeat it many times.